File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/BaseActivity.java
package net.kdt.pojavlaunch;

import android.content.*;
import android.content.res.Configuration;
import android.os.*;

import androidx.annotation.NonNull;
import androidx.appcompat.app.*;
import net.kdt.pojavlaunch.utils.*;

import static net.kdt.pojavlaunch.prefs.LauncherPreferences.PREF_IGNORE_NOTCH;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(LocaleUtils.setLocale(newBase));
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        LocaleUtils.setLocale(this);
        Tools.setInsetsMode(this, setFullscreen(), shouldIgnoreNotch());
        Tools.getDisplayMetrics(this);
    }

    /** @return Whether the activity should be set as a fullscreen one */
    public boolean setFullscreen(){
        return true;
    }


    @Override
    public void startActivity(Intent i) {
        super.startActivity(i);
    }

    @Override
    protected void onResume() {
        super.onResume();
        Tools.checkStorageInteractive(this);
    }

    @Override
    protected void onPostResume() {
        super.onPostResume();
        Tools.setInsetsMode(this, setFullscreen(), shouldIgnoreNotch());
        Tools.getDisplayMetrics(this);
    }

    @Override
    public void onMultiWindowModeChanged(boolean isInMultiWindowMode) {
        super.onMultiWindowModeChanged(isInMultiWindowMode);
        onMultiWindowModeChangedInner();
    }

    @Override
    public void onMultiWindowModeChanged(boolean isInMultiWindowMode, @NonNull Configuration newConfig) {
        super.onMultiWindowModeChanged(isInMultiWindowMode, newConfig);
        onMultiWindowModeChangedInner();
    }

    private void onMultiWindowModeChangedInner() {
        boolean wantsFullscreen = setFullscreen();
        if(wantsFullscreen) {
            Tools.setInsetsMode(this, true, shouldIgnoreNotch());
        }
    }

    /** @return Whether or not the notch should be ignored */
    protected boolean shouldIgnoreNotch(){
        return PREF_IGNORE_NOTCH;
    }
}

File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/BaseActivity.kt
package net.kdt.pojavlaunch

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import net.kdt.pojavlaunch.utils.LocaleUtils

import net.kdt.pojavlaunch.prefs.LauncherPreferences.PREF_IGNORE_NOTCH

abstract class BaseActivity : AppCompatActivity() {

    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(LocaleUtils.setLocale(newBase))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        LocaleUtils.setLocale(this)
        Tools.setInsetsMode(this, setFullscreen(), shouldIgnoreNotch())
        Tools.getDisplayMetrics(this)
    }

    open fun setFullscreen(): Boolean {
        return true
    }

    override fun startActivity(i: Intent) {
        super.startActivity(i)
    }

    override fun onResume() {
        super.onResume()
        Tools.checkStorageInteractive(this)
    }

    override fun onPostResume() {
        super.onPostResume()
        Tools.setInsetsMode(this, setFullscreen(), shouldIgnoreNotch())
        Tools.getDisplayMetrics(this)
    }

    protected open fun shouldIgnoreNotch(): Boolean {
        return PREF_IGNORE_NOTCH
    }
}
